aiocache/plugins.py

```diff
@@ -96,3 +96,49 @@ async def post_multi_get(self, client, keys, took=0, ret=None, **kwargs):
         client.hit_miss_ratio["hit_ratio"] = (
             client.hit_miss_ratio["hits"] / client.hit_miss_ratio["total"]
         )
+
+
+class LimitLengthPlugin(BasePlugin):
+    """
+    Limits the number of entries that the cache may contain.
+
+    Example usage:
+      c = Cache(cache_class=Cache.MEMORY, ttl=300, plugins=[LimitLength(max_length=2000)])
+
+    Caveats:
+    * Only works with the SimpleMemoryCache backend.
+    * Entries are trimmed AFTER the new entry is added, so the cache may temporarily contain more entries than the specified limit.
+    """
+
+    def __init__(self, max_length: int = 1000):
+        assert int(max_length) == max_length
+        assert max_length >= 0
+        self.max_length = int(max_length)
+
+    def _check_support(self, client: SimpleMemoryCache):
+        """
+        Checks if the client is of the correct type.
+        TODO: Ideally, this should be checked only once when the cache is created.
+        """
+        if not isinstance(client, SimpleMemoryCache):
+            raise NotImplementedError(
+                "LimitLength plugin only works with the SimpleMemoryCache backend."
+            )
+
+    async def post_set(self, client: SimpleMemoryCache, *args, **kwargs):
+        """
+        Drop extra records that exceed configured capacity.
+        """
+        self._check_support(client)
+        current_length = len(client._cache)
+        if current_length > self.max_length:
+            # Sort by time of expiry. Delete the entries nearest to expiration first.
+            a = list((client._handlers[k].when(), k)
+                     for k in client._handlers.keys())
+            a.sort()
+            for _, k in a[:current_length - self.max_length]:
+                await client.delete(k)
```

Comment:
No problem. I'll try a different approach. Please close this pull request.

Reply:
As mentioned in the issue, maybe it's worth creating a custom backend with Theine?
You could then publish that backend somewhere and add it to the README here.

The memory backend here is rather simple and probably best for testing things without needing to have a redis/memcache server available, rather than being relied upon as a production backend.

Reply:
Yes, handling record expiration efficiently is non-trivial, using a dedicated tool seems sensible. I have still not made sense of most of the codebase - the unit test part, for example - so I'm not at a position to write a backend for that. We need just the basic functionality for an API we are deploying at my employer, and he is fine with me contributing it back to the project, hence the merge request. I agree that it does not pass the bar for a production library like aiocache. I am in crunch mode for more than a week now, stamina running really low.